exonum-java-binding-qa-service/src/test/java/com/exonum/binding/qaservice/QaServiceImplIntegrationTest.java

```diff
@@ -351,13 +344,104 @@ void getActualConfigurationBeforeInit() {
   @Test
   @RequiresNativeLibrary
   void getActualConfiguration() {
+    withNodeFake(() -> {
+      Throwable t = assertThrows(RuntimeException.class, () -> service.getActualConfiguration());
+      assertThat(t.getMessage()).contains("Couldn't not find any config for"
+          + " height 0, that means that genesis block was created incorrectly.");
+    });
+  }
+
+  @Test
+  @RequiresNativeLibrary
+  void getBlockTransactionsByBlockId() {
+    withNodeFake(() -> {
+      Throwable t = assertThrows(RuntimeException.class, () -> service.getBlockTransactions(0L));
+      assertThat(t.getMessage()).contains(NO_GENESIS_BLOCK_ERROR_MESSAGE);
+    });
+  }
+
+  @Test
+  @RequiresNativeLibrary
+  void getTxMessages() {
+    withNodeFake(() -> {
+      Map<HashCode, TransactionMessage> txMessages = service.getTxMessages();
+      assertThat(txMessages).isEmpty();
+    });
+  }
+
+  @Test
+  @RequiresNativeLibrary
+  void getTxResults() {
+    withNodeFake(() -> {
+      Map<HashCode, TransactionResult> txResults = service.getTxResults();
+      assertThat(txResults).isEmpty();
+    });
+  }
+
+  @Test
+  @RequiresNativeLibrary
+  void getTxResult() {
+    withNodeFake(() -> {
+      HashCode messageHash = HashCode.fromBytes(createPrefixed(bytes(0x00), 32));
+      Optional<TransactionResult> txResult = service.getTxResult(messageHash);
+      assertThat(txResult).isEmpty();
+    });
+  }
+
+  @Test
+  @RequiresNativeLibrary
+  void getTxLocations() {
+    withNodeFake(() -> {
+      Map<HashCode, TransactionLocation> txLocations = service.getTxLocations();
+      assertThat(txLocations).isEmpty();
+    });
+  }
+
+  @Test
+  @RequiresNativeLibrary
+  void getTxLocation() {
+    withNodeFake(() -> {
+      HashCode messageHash = HashCode.fromString("ab");
+      Optional<TransactionLocation> txLocation = service.getTxLocation(messageHash);
+      assertThat(txLocation).isEmpty();
+    });
+  }
+
+  @Test
+  @RequiresNativeLibrary
+  void getBlocks() {
+    withNodeFake(() -> {
+      Map<HashCode, Block> blocks = service.getBlocks();
+      assertThat(blocks).isEmpty();
+    });
+  }
+
+  @Test
+  @RequiresNativeLibrary
+  void getBlock() {
+    withNodeFake(() -> {
+      HashCode blockId = HashCode.fromString("ab");
+      Optional<Block> block = service.getBlock(blockId);
+      assertThat(block).isEmpty();
+    });
+  }
+
+  @Test
+  @RequiresNativeLibrary
+  void getLastBlock() {
+    withNodeFake(() -> {
+      Exception e = assertThrows(RuntimeException.class, () -> service.getLastBlock());
+      assertThat(e).hasMessageContaining("An attempt to get the `last_block` during creating the genesis block.");
+    });
+  }
+
+  /** Runs a test with a service with a node fake set. */
+  private void withNodeFake(Runnable test) {
```

Comment:
BTW, another alternative that allows to share some setup/teardown code is a `@Nested` test class that has its own `@BeforeEach`/`@AfterEach`:
```java
@Nested
class WithNodeFake

  // @BeforeEach & AfterEach

  @Test
  void fooTest() {
    // No need to put your test code in a lambda — enjoy an initialized service
  }
```

Reply:
No, thanks, it wasn't a suggestion for a change but an alternative approach that might be slightly better in terms of readability.